src/components/Header.jsx

```diff
@@ -0,0 +1,47 @@
+import { Link, NavLink } from "react-router-dom"
+import { useSelector } from 'react-redux'
+
+import '../styles/header.scss'
+
+const Header = ({ searchMovies }) => {
+  
+  const { starredMovies } = useSelector((state) => state.starred)
+
+  return (
+    <header>
+      <Link to="/" data-testid="home" onClick={() => searchMovies('')}>
```

Comment:
Arrow functions in JSX, especially when utilized as event handlers or as props, could lead to unnecessary re-renders and performance bottlenecks. One effective alternative is leveraging the use callback hook to define and memoize functions, ensuring they're created only once.